UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs

```diff
@@ -58,7 +59,16 @@ private void UpdateMe()
 					return;
 				}
 				mainThreadTimer.Restart();
-				RunTick();
+				try
+				{
+					RunTick();
+				}
+				catch (Exception e)
+				{
+					Logger.LogError(e.ToString(), Category.Threading);
+					throw;
```

Comment:
can you do this? Don't you have to say what exception to throw?

Reply:
if I put e on the end of it, it just recommends says rethrowing
![image](https://user-images.githubusercontent.com/27101670/161613641-4a323707-21fd-4e83-b3b7-10db002c1704.png)

![image](https://user-images.githubusercontent.com/27101670/161613706-61eefed1-ed21-449d-95b3-c02218013e78.png)

Reply:
ah cool, it must infere it from the context then